### ISSUE TITLE
dockerng.running reports an odd diff for entrypoint changes

### ISSUE THREAD

**Issue by hjc** [saltstack/salt#31257]
If you use `dockerng.running` and change the `entrypoint` for your Docker container, then the state reports the diff in a very odd format. It looks like each item in the entrypoint is sorted alphabetically and that is chosen. This is jarring because the `entrypoint` key takes in a list of various command line arguments/flags that are joined together in the order they are specified in.

Here is the entrypoint we are using: 

```
app-worker:
  dockerng.running:
    - image: company/app:{{ pillar['sw2.appenv'] }}
    - network_mode: host
    - entrypoint:
      - celery
      - -A
      - application.celery_app
      - worker
      - --loglevel=info
      - --autoscale={{- pillar['app.worker.max_procs'] -}},{{-
                        pillar['app.worker.min_procs'] -}}
      - -n
      - worker
    - environment:
      - APP_PORT: "{{ 5000 }}"
      - WORKER_COUNT: "1"
      - REDIS_HOST: localhost
      - CELERYD_MAX_TASKS_PER_CHILD: "100"
      - RABBITMQ_DEFAULT_HOST: localhost
      - RABBITMQ_DEFAULT_PASS: password
      - MYSQL_DEFAULT_HOST: localhost
      - MYSQL_DEFAULT_USER: user
      - MYSQL_DEFAULT_PASS: password
      - DATABASE_PORT: "3306"
```

And whenever we change something in the entrypoint, the diff we get resembles:

```
----------
          ID: app-worker
    Function: dockerng.running
      Result: True
     Comment: Container 'app-worker' changed state.. Container 'app-worker' was replaced.
     Started: 16:28:01.454594
    Duration: 10256.197 ms
     Changes:   
              ----------
              added:
                  ----------
                  Id:
                      b95f2585b4948f40f45e0d6d18bb9d17cd1d0f47588ea02f37d6d066f2fa6c2f
                  Name:
                      app-worker
                  Time_Elapsed:
                      0.143102169037
                  Warnings:
                      None
              diff:
                  ----------
                  entrypoint:
                      ----------
                      new:
                          - --autoscale=2,1
                          - --loglevel=info
                          - -A
                          - -n
                          - application.celery_app
                          - celery
                          - worker
                          - worker
                      old:
                          - --autoscale=2,1- -n
                          - --loglevel=info
                          - -A
                          - application.celery_app
                          - celery
                          - worker
                          - worker
              removed:
                  - 1ff83a6915ba6e7cb8c2d28f3d33d631be397a7007d3a32b3cab6805e0fdf90c
              state:
                  ----------
                  new:
                      running
                  old:
                      stopped
```

What I expect to be printed out:

```
----------
          ID: app-worker
    Function: dockerng.running
      Result: True
     Comment: Container 'app-worker' changed state.. Container 'app-worker' was replaced.
     Started: 16:28:01.454594
    Duration: 10256.197 ms
     Changes:   
              ----------
              added:
                  ----------
                  Id:
                      b95f2585b4948f40f45e0d6d18bb9d17cd1d0f47588ea02f37d6d066f2fa6c2f
                  Name:
                      app-worker
                  Time_Elapsed:
                      0.143102169037
                  Warnings:
                      None
              diff:
                  ----------
                  entrypoint:
                      ----------
                      new:
                          - celery
                          - -A
                          - application.celery_app
                          - worker
                          - --loglevel=info
                          - --autoscale=2,1
                          - -n
                          - worker
                      old:
                          - celery
                          - -A
                          - application.celery_app
                          - worker
                          - --loglevel=info
                          - --autoscale=2,1- -n
                          - worker
              removed:
                  - 1ff83a6915ba6e7cb8c2d28f3d33d631be397a7007d3a32b3cab6805e0fdf90c
              state:
                  ----------
                  new:
                      running
                  old:
                      stopped
```

I expect this because that is the order in which I defined my entrypoint pieces. It's also relevant because the order in which I pass my arguments to my entrypoint command generally matters, and this causes confusion (and a bit of terror as I fear my deploy is broken!).

Thanks in advance!

`salt --versions-report` output:

```
Salt Version:
           Salt: 2015.8.5

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 4.1.5-x86_64-linode61
         system: Ubuntu 14.04 trusty
```

**Comment by jfindlay** [comment]
@hjc1710, thanks for the report.

**Comment by terminalmage** [comment]
This was fixed in 2017.7.0, marking as closed.
